commands/command_track.go

```diff
@@ -74,32 +78,43 @@ ArgsLoop:
 			}
 		}
 
-		encodedArg := strings.Replace(pattern, " ", "[[:space:]]", -1)
-		_, err := attributesFile.WriteString(fmt.Sprintf("%s filter=lfs diff=lfs merge=lfs -text\n", encodedArg))
-		if err != nil {
-			Print("Error adding path %s", pattern)
-			continue
+		if !trackDryRunFlag {
+			encodedArg := strings.Replace(pattern, " ", "[[:space:]]", -1)
+			_, err := attributesFile.WriteString(fmt.Sprintf("%s filter=lfs diff=lfs merge=lfs -text\n", encodedArg))
+			if err != nil {
+				Print("Error adding path %s", pattern)
+				continue
+			}
 		}
 		Print("Tracking %s", pattern)
 
-		// Make sure any existing git tracked files have their timestamp updated
-		// so they will now show as modifed
-		// note this is relative to current dir which is how we write .gitattributes
-		// deliberately not done in parallel as a chan because we'll be marking modified
-		gittracked, err := git.GetTrackedFiles(pattern)
-		if err != nil {
-			LoggedError(err, "Error getting git tracked files")
-			continue
-		}
-		now := time.Now()
-		for _, f := range gittracked {
-			err := os.Chtimes(f, now, now)
+		if !trackNoTouchFlag || trackDryRunFlag {
```

Comment:
If `--no-touch` is used then we should still be checking whether there are matching files which are already added to git, because failing to do this will cause nasty historical problems if the `.gitattributes` is committed without also re-committing the affected tracked files. 

At an absolute minimum it should warn the user that this is the case, but really I think it should abort. Not `touch`ing the files just causes too many unfixable issues (without history re-writing) otherwise.